src/main/java/net/dries007/tfc/util/json/VeinTypeJson.java

```diff
@@ -68,6 +86,49 @@ public VeinType deserialize(JsonElement json, Type typeOfT, JsonDeserializationC
             }
         });
 
+        // Parse loose rock
+        ItemStack looseRock = ItemStack.EMPTY;
+        if (JsonUtils.hasField(jsonObject, "loose"))
+        {
+            ResourceLocation looseResource = new ResourceLocation(JsonUtils.getString(jsonObject, "loose"));
+            // try parsing small ore first
+            Ore looseOre = TFCRegistries.ORES.getValue(looseResource);
+            Item looseItem = null;
+            if (looseOre != null)
+            {
+                looseItem = ItemSmallOre.get(looseOre);
+            }
+            if (looseItem == null)
+            {
+                // Try parsing item instead
+                // todo remove meta in 1.15+
+                int metaLoose = JsonUtils.getInt(jsonObject, "looseMeta", 0);
+                looseItem = ForgeRegistries.ITEMS.getValue(looseResource);
+                if (looseItem == null)
+                {
+                    // Block?
+                    Block looseBlock = ForgeRegistries.BLOCKS.getValue(looseResource);
+                    // Looks like forge parse anything to air if nothing is found
+                    if (looseBlock == null || looseBlock == Blocks.AIR)
+                    {
+                        throw new JsonParseException("Unable to parse loose rock " + looseResource + ". No registered small ore, item or block found.");
+                    }
+                    else
+                    {
+                        looseRock = new ItemStack(looseBlock, 1, metaLoose);
+                    }
```

Comment:
This whole sequence is unused, because you construct an `ItemStack` anyway, which will either:
a. convert the `Block` to it's respective `Item` (which, if it exists, would've been caught from the item query above), or
b. if the block doesn't exist, it will return an empty item stack.